server/src/main/java/com/objectcomputing/checkins/services/role/RoleServicesImpl.java

```diff
@@ -20,46 +21,44 @@ public RoleServicesImpl(RoleRepository roleRepo,
         this.memberRepo = memberRepo;
     }
 
-    public Role save(Role role) {
+    public Role save(@NotNull Role role) {
         Role roleRet = null;
-        if (role != null) {
-            final UUID memberId = role.getMemberid();
-            final RoleType roleType = role.getRole();
-            if (roleType == null || memberId == null) {
-                throw new RoleBadArgException(String.format("Invalid role %s", role));
-            } else if (role.getId() != null) {
-                throw new RoleBadArgException(String.format("Found unexpected id %s for role", role.getId()));
-            } else if (memberRepo.findById(memberId).isEmpty()) {
-                throw new RoleBadArgException(String.format("Member %s doesn't exist", memberId));
-            } else if (roleRepo.findByRoleAndMemberid(roleType, role.getMemberid()).isPresent()) {
-                throw new RoleBadArgException(String.format("Member %s already has role %s", memberId, roleType));
-            }
-
-            roleRet = roleRepo.save(role);
+        final UUID memberId = role.getMemberid();
+        final RoleType roleType = role.getRole();
+
+        if (roleType == null || memberId == null) {
+            throw new BadArgException(String.format("Invalid role %s", role));
+        } else if (role.getId() != null) {
+            throw new BadArgException(String.format("Found unexpected id %s for role", role.getId()));
+        } else if (memberRepo.findById(memberId).isEmpty()) {
+            throw new BadArgException(String.format("Member %s doesn't exist", memberId));
+        } else if (roleRepo.findByRoleAndMemberid(roleType, role.getMemberid()).isPresent()) {
+            throw new BadArgException(String.format("Member %s already has role %s", memberId, roleType));
         }
+
+        roleRet = roleRepo.save(role);
```

Comment:
We could remove the variable `roleRet` and return this directly

Reply:
done